sdk/rust/reference.mdx

```diff
@@ -1,4 +1,275 @@
+
+## Local Development
+
+You can work locally using an SQLite file using `new_local`:
+
+```rust
+use libsql::Builder;
+
+let mut db = Builder::new_local("local.db").build().await.unwrap();
+let conn = db.connect().unwrap();
+```
+
+## Embedded Replicas
+
+You can work with embedded replicas using `new_remote_replica` that can sync from the remote URL and delegate writes to the remote primary database:
+
+```rust
+use libsql::Builder;
+
+let url = env::var("LIBSQL_URL").expect("LIBSQL_URL must be set");
+let token = env::var("LIBSQL_AUTH_TOKEN").unwrap_or_default();
+
+let mut db = Builder::new_remote_replica("local.db", &url, &token).build().await.unwrap();
+let conn = db.connect().unwrap();
+```
+
+### Manual Sync
+
+The `sync` function allows you to sync manually the local database with the remote counterpart:
+
+```rust
+use libsql::Builder;
+
+let url = env::var("LIBSQL_URL").expect("LIBSQL_URL must be set");
+let token = env::var("LIBSQL_AUTH_TOKEN").unwrap_or_default();
+
+let mut db = Builder::new_remote_replica("local.db", &url, &token).build().await.unwrap();
+
+let conn = db.connect().unwrap();
+
+db.sync().await.unwrap(); // Call sync manually to update local database
+```
+
+<Info>
+
+If you require full control over how frames get from your instance of `sqld` (libSQL Server), you can do this using `new_local_replica` and `sync_frames`. Reach out to us [on Discord](https://discord.com/invite/4B5D7hYwub) if you want to learn more.
+
+</Info>
+
+### Periodic Sync
+
+The `periodic_sync` function allows you to set an interval for automatic synchronization of the database in the background:
+
+```rust
+use libsql::Builder;
+use std::time::Duration;
+
+let url = env::var("LIBSQL_URL").expect("LIBSQL_URL must be set");
+let token = env::var("LIBSQL_AUTH_TOKEN").unwrap_or_default();
+
+let mut db = Builder::new_remote_replica("local.db", &url, &token)
+  .periodic_sync(Duration::from_secs(300)) // Sync every 5 minutes
+  .build()
+  .await.unwrap();
+
+let conn = db.connect().unwrap();
+```
+
+### Read Your Own Writes
+
+The `read_your_writes` function configures the database connection to ensure that writes are immediately visible to subsequent read operations initiated by the same connection. This is enabled by default and is particularly important in distributed systems to ensure consistency from the perspective of the writing process.
+
+You can disable this behavior by passing `false` to the function:
+
+```rust
+use libsql::Builder;
+
+let url = env::var("LIBSQL_URL").expect("LIBSQL_URL must be set");
+let token = env::var("LIBSQL_AUTH_TOKEN").unwrap_or_default();
+
+let mut db = Builder::new_remote_replica("local.db", &url, &token)
+  .read_your_writes(false) // Disable reading your own writes
+  .build()
+  .await.unwrap();
+
+let conn = db.connect().unwrap();
+```
+
+## Encryption
+
+To enable encryption on a SQLite file (`new_local` or `new_remote_replica`), pass the `encryption_config`:
```

Comment:
make sure to mention this requires enabling the `encryption` feature which is disabled by default.

Reply:
Ah, @LucioFranco this is a great catch. I'm going to further tweak the examples to show how to enable the features with `features = ["encryption"]` etc.